File: Relative_Sort_Array/solution.best2.py
class Solution:
  def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:
    countMap = [0] * 1001
    
    for v in arr1: countMap[v] += 1
      
    i = 0
    for v in arr2:
      for j in range(countMap[v]):
        arr1[i] = v
        i += 1
      countMap[v] = 0
    
    for v in range(1001):
      for j in range(countMap[v]):
        arr1[i] = v
        i += 1
    
    return arr1

File: Relative_Sort_Array/solution.best2.java
class Solution {
  public int[] relativeSortArray(int[] arr1, int[] arr2) {
    int countMap[] = new int[1001];
    for (int v : arr1) countMap[v]++;
    
    int i = 0;
    for (int v : arr2) {
      for (int j = countMap[v]; j > 0; j--) arr1[i++] = v;
      countMap[v] = 0;
    }
    
    for (int v = 0; v < 1001; v++) {
      for (int j = countMap[v]; j > 0; j--) arr1[i++] = v;
    }
    
    return arr1;
  }
}
